pkg/controller/spinnakervalidating/spinnakervalidating_controller.go

```diff
@@ -30,19 +38,129 @@ var log = logf.Log.WithName("spinvalidate")
 
 // Add adds the validating admission controller
 func Add(m manager.Manager) error {
-	//ns, err := k8sutil.GetOperatorNamespace()
-	_, err := k8sutil.GetOperatorNamespace()
+	// Determine environment
+	ns, name, err := getOperatorNameAndNamespace()
 	if err != nil {
 		return err
 	}
 
+	// Create Kubernetes service for listening to requests from API server
+	rawClient := kubernetes.NewForConfigOrDie(m.GetConfig())
+	port := 9876
+	err = deployWebhookService(ns, name, port, rawClient)
+	if err != nil {
+		return err
+	}
+
+	// Create or get certificates
+	c, err := getCertContext(ns, name)
+	if err != nil {
+		return err
+	}
+
+	// Register webhook server
 	hookServer := m.GetWebhookServer()
-	hookServer.Register("/validate-v1alpha1-spinnakerservice", &webhook.Admission{Handler: &spinnakerValidatingController{}})
-	return nil
+	hookServer.CertDir = c.certDir
+	hookServer.Port = port
+	gv := SpinnakerServiceBuilder.GetGroupVersion()
+	path := fmt.Sprintf("/validate-%s-spinnakerservice", gv.Version)
```

Comment:
Use the `kind` from `SpinnakerServiceBuilder`. I'd just use from the controller-runtime's webhook.go:
```
func generateValidatePath(gvk schema.GroupVersionKind) string {
	return "/validate-" + strings.Replace(gvk.Group, ".", "-", -1) + "-" +
		gvk.Version + "-" + strings.ToLower(gvk.Kind)
}
```

Reply:
There's no `kind` in `SpinnakerServiceBuilder`, just a `GroupVersion` object